### DISCUSSION TITLE
Can't log into Hive after upgrade

### DISCUSSION THREAD

**Discussion by B3DTech** [Security-Onion-Solutions/securityonion#2501] [Unsupported Versions]
Upgraded to 2.3.21 and everything seems to work except the Hive.  When I go to open it, I get a repeating, never ending error:  

App: user init not found

Not seeing anything in the logs for it.

**Comment by B3DTech** [comment]
More fun - I was finally able to log out of the hive.  Can't log back in with the account I've been using since day 1.  I created a new user using so-user-add and while I can log in with it, I get the following errors:

StatSrv: Invalid Search Query
SearchSrv: SearchErr
AuditSrv: Undefined

and I have no access to anything.

**Reply by TOoSmOotH** [reply]
Can you check the ES index? ES was update from 6 to 7 for thehive

**Reply by B3DTech** [reply]
The ES index would give an authentication error when trying to sign in?

**Reply by TOoSmOotH** [reply]
That is where the creds are kept in the hive es instance.

**Reply by marsianinz** [reply]
> More fun - I was finally able to log out of the hive. Can't log back in with the account I've been using since day 1. I created a new user using so-user-add and while I can log in with it, I get the following errors:
> 
> StatSrv: Invalid Search Query
> SearchSrv: SearchErr
> AuditSrv: Undefined
> 
> and I have no access to anything.

I get the same problem after update. Have any solve?

**Comment by B3DTech** [comment]
```
[2020-12-30T21:19:38,335][INFO ][org.elasticsearch.node.Node] initialized
[2020-12-30T21:19:38,335][INFO ][org.elasticsearch.node.Node] starting ...
[2020-12-30T21:19:38,453][INFO ][org.elasticsearch.transport.TransportService] publish_address {172.17.0.20:9500}, bound_addresses {0.0.0.0:9500}
[2020-12-30T21:19:38,651][INFO ][org.elasticsearch.cluster.coordination.Coordinator] cluster UUID [VV8bDop3RFey5WfQzjSaSg]
[2020-12-30T21:19:38,754][INFO ][org.elasticsearch.cluster.service.MasterService] elected-as-master ([1] nodes joined)[{so-thehive-es}{hk00l9qyR3yE6AI12MytuA}{ACutFbEfS6KSVRga3TE3SA}{172.17.0.20}{172.17.0.20:9500}{dimr} elect leader, _BECOME_MASTER_TASK_, _FINISH_ELECTION_], term: 2, version: 25, delta: master node changed {previous [], current [{so-thehive-es}{hk00l9qyR3yE6AI12MytuA}{ACutFbEfS6KSVRga3TE3SA}{172.17.0.20}{172.17.0.20:9500}{dimr}]}
[2020-12-30T21:19:38,800][INFO ][org.elasticsearch.cluster.service.ClusterApplierService] master node changed {previous [], current [{so-thehive-es}{hk00l9qyR3yE6AI12MytuA}{ACutFbEfS6KSVRga3TE3SA}{172.17.0.20}{172.17.0.20:9500}{dimr}]}, term: 2, version: 25, reason: Publication{term=2, version=25}
[2020-12-30T21:19:38,818][INFO ][org.elasticsearch.http.AbstractHttpServerTransport] publish_address {172.17.0.20:9400}, bound_addresses {0.0.0.0:9400}
[2020-12-30T21:19:38,818][INFO ][org.elasticsearch.node.Node] started
[2020-12-30T21:19:38,870][INFO ][org.elasticsearch.gateway.GatewayService] recovered [4] indices into cluster_state
[2020-12-30T21:19:39,912][INFO ][org.elasticsearch.cluster.routing.allocation.AllocationService] Cluster health status changed from [RED] to [YELLOW] (reason: [shards started [[cortex_4][2]]]).


green  open cortex_4    8Mxi1p0MTWqqASMtGF4ezg 5 0      4    0 27.4kb 27.4kb
green  open cortex_5    LyB-46zERqqKSlSMwodwTQ 5 0      4    0   29kb   29kb
green  open the_hive_15 AFHFDMk_RIadwHKklm4xbA 5 0 177148 7141 53.3mb 53.3mb
yellow open the_hive_16 h_uBe6QFTvOEl-vGVG9vWQ 1 1      5    0 54.6kb 54.6kb
```

**Comment by cowbe0x004** [comment]
User management doesn't seem to work any more. I can still login to SOC but not thehive and fleet. Tried creating new users, same thing.
And is there a way to delete a user?

**Reply by defensivedepth** [reply]
@cowbe0x004 are you still having this issue? What version are you on? If you try to add a new user by running `so-user-add`, what is the output?

**Comment by facyber** [comment]
Hello, I couldn't access The Hive after fresh install, I was getting authentication error. By documentation it should be the same credentials for accessing Security Onion Console.

So I added new hive user and enabled it:
```
so-thehive-user-add <$USERNAME>
so-thehive-user-enable <$USERNAME>
```

And login the worked for me, but then I saw there was already an user hiveadmin. I believe there is some issue with creating The Hive user during installation maybe.

**Comment by UMHB-InfoSec** [comment]
I am having this issue on a new install as well.  I can log into SOC but not Fleet or The Hive.  If I try to create the user that I use to log into SOC in The Hive with the commands:

```
so-thehive-user-add <$USERNAME>
so-thehive-user-enable <$USERNAME>
```

I get the error:

```
so-thehive-user-add <my user>
Enter new password:
Unable to add user to TheHive; user might already exist
{"type":"AuthenticationError","message":"Authentication failure"}
```
Any ideas?

**Reply by thegreekman** [reply]
Im getting the same issue, did you ever find a resolution?

**Reply by UMHB-InfoSec** [reply]
I am still working on it, but I did find out some interesting info.

The password I used has spaces and it seems that spaces are not escaped properly in the install script.  For fleet I was able to log in using the first part of the password I set and omitting the first space and everything beyond it.  

EG: The password could be "battery horse staple correct"  and I was able to log in to fleet using "battery"

This aligns well with something else I noticed during the OS install when asked to provide a password for the OS user if the provided password contains spaces the password consistency check will fail, probably also an escape issue.

However, the same trick DID NOT work on The Hive ... sad day for me.

I am still looking into the error there.  

`Unable to add user to TheHive; user might already exist
{"type":"AuthenticationError","message":"Authentication failure"}`

Some of my google-ing leads to posts that are very similar that mention a bad API token but I haven't had the chance to track that one down yet.

**Reply by thegreekman** [reply]
Thanks for the response. I really don't want to have to re-install all of security onion just to fix this issue, I wonder if we can get a new api token or something

**Reply by UMHB-InfoSec** [reply]
I don't know how to generate one yet, I'm still learning the new SO architecture.  If you figure it out please post how here and I'll do the same if I figure it out.

**Reply by thegreekman** [reply]
I found a typo in the so-thehive-user-add and so-thehive-user-enable scripts. 
resp=$(curl -sk -XPOST -H "Authorization: Bearer $THEHIVE_KEY" -H "Content-Type: application/json" -L "https://$**THEHVIE**_API_URL/user" -d "{\"login\" : \"$THEHIVE_USER\",\"name\" : \"$THEHIVE_USER\",\"roles\" : [\"read\",\"alert\",\"write\",\"admin\"],\"preferences\" : \"{}\",\"password\" : \"$THEHIVE_PASS\"}")


"THEHIVE_API" is misspelled in both scripts, but this is irrelevant since the typo occurs at the top where the value is set as well.


I spun up a new SO on a test system and I was able to add a user. However, after I locked the main admin account, I received the authentication error again.

At least for me, my problem is due to the fact that the API token is no longer valid, since I locked the account associated to the API token that SO created initially. I need to find a way to create a new one in order for me to fix this problem.

**Reply by UMHB-InfoSec** [reply]
So I found this entry in the hive's github issue tracker:

https://github.com/TheHive-Project/TheHive/issues/233

After following it I was able to reset the password and get in!

I followed the instructions from this post:

https://github.com/TheHive-Project/TheHive/issues/233#issuecomment-607288708

Basically, once you know your hive instance id you can XDELETE it and when you refrersh the page it will prompt you to setup a new admin account.

**Reply by thegreekman** [reply]
So instead of localhost I use the IP address of my management interface? and should Include the underscores in "the_hive"?

**Reply by thegreekman** [reply]
Nevermind, got it to work!

I used the following: curl -XDELETE http://127.0.0.1:9400/the_hive_16 
                               curl -XDELETE http://127.0.0.1:9400/the_hive_17

For some reason, I had 2 instances. Thank you so much for this solution. Truly a life savor!

**Comment by mtnsec** [comment]
I'm getting the same error as the original poster:

App: user init not found

I tried adding a new user and I'm getting the same error as the most recent poster:

```
Unable to add user to TheHive; user might already exist
{"type":"com.sksamuel.elastic4s.http.JavaClientExceptionWrapper","message":"java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]"}
```
However, their solution doesn't work for me. I created the script to search for the hive instance ID but it appears I have none:

Script:
```
#!/bin/bash
for i in {1..20}
do
        curl http://127.0.0.1:9200/the_hive_$i
        echo ""
        echo ""
done
```
Every response in the loop is:

`curl: (52) Empty reply from server`

So, how do I fix it?

**Reply by UMHB-InfoSec** [reply]
Try using port 9400 instead of 9200?  9400 is the port Hive is using by default in SO.  I don't know what is running 9200, it could be nothing or something that doesn't respond to the query which would explain the timeout error.

**Reply by mtnsec** [reply]
Ah, I think 9200 is elasticsearch.  I just copied the script from here:  https://github.com/TheHive-Project/TheHive/issues/233#issuecomment-607288708

9400 isn't working though:  

`curl: (7) Failed to connect to localhost port 9400: Connection refused`

so-status shows the hive is running:
```
    so-thehive ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-thehive-es ---------------------------------------------------------------------------------------------------------------------------------- [ OK ] 
```

**Reply by UMHB-InfoSec** [reply]
Forgive me for asking, but are you running the script directly on the SO master server?  I have had issues similar to what you are saying if I try to run the curl from a remote station.

Just to make sure I am communicating correctly, when I say directly on the server I don't mean, necessarily, on the physical console; an SSH session is perfectly fine too.

**Reply by mtnsec** [reply]
Hi!  Yes I was on the master. I ran so-thehive-restart to see if it would fix the connection refused error and it did. The script showed I had 3 hive instances!

After deleting them and refreshing the web interface, I got the message:

"TheHive needs to update its database"

So I clicked the "Update database" button.

And it appears to do nothing.  Refreshing again brings me back to where I started with the "App: user init not found" error. :(

I can delete the instance again, and it puts me back at the update database screen.

**Reply by mtnsec** [reply]
Looks like 2nd time was the charm?  I tried "update database" again after deleting the instance again and the second time it worked, and prompted me to create the user as expected.  

Thanks for your help!!!

**Reply by UMHB-InfoSec** [reply]
WOOT!

**Comment by Mastadamus** [comment]
Has anyone found a solution for this? i'm getting the exact same error after upgrading via soup. I just get a flood of "App: user init not found"

**Reply by UMHB-InfoSec** [reply]
Did you try this?

https://github.com/Security-Onion-Solutions/securityonion/discussions/2501#discussioncomment-526209

It worked for me and another user.

**Reply by Mastadamus** [reply]
Thanks. I did but I seemed to have deleted or lost the original account as I couldn't login with my soc account anymore.  I created a new one. Then I had same problem with cortex. I did same steps for cortex. Thought it was working.  Then I realized analyzers enabled in cortex wouldn't show in the hive.  I finally reverted back to a pre 2.3.50 snapshot and reupgraded straight to the 2.3.50 hot fix. It now works.
